server/src/main/java/org/apache/uniffle/server/ShuffleFlushManager.java

```diff
@@ -111,26 +114,28 @@ public ShuffleFlushManager(ShuffleServerConf shuffleServerConf, String shuffleSe
     processEventThread.setName("ProcessEventThread");
     processEventThread.setDaemon(true);
     processEventThread.start();
-    // todo: extract a class named Service, and support stop method
-    Thread thread = new Thread("PendingEventProcessThread") {
-      @Override
-      public void run() {
-        for (; ; ) {
-          try {
-            processPendingEvents();
-            processPendingEventIndex = (processPendingEventIndex + 1) % 1000;
-            if (processPendingEventIndex == 0) {
-              // todo: get sleep interval from configuration
-              Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
-            }
-          } catch (Exception e) {
-            LOG.error(getName() + " happened exception: ", e);
+    pendingEventProcesser = Executors.newSingleThreadScheduledExecutor(
+            ThreadUtils.getThreadFactory("PendingEventProcesser-%d"));
+    long sleepInterval = shuffleServerConf.get(ShuffleServerConf.PENDING_EVENT_CHECK_INTERVAL);
+    long memoryCapacity = shuffleServerConf.getSizeAsBytes(ShuffleServerConf.SERVER_BUFFER_CAPACITY);;
+    long memoryThreshold = Math.max(memoryCapacity / 10, 1024L * 1024 * 1024);
```

Comment:
Why do we divide by 10?

Reply:
Use 10 seconds to release all buffer, Maybe bigger than 10 is better, I am observing in our production environment